genesyscloud/resource_genesyscloud_externalcontacts_contact.go

```diff
@@ -0,0 +1,812 @@
+	}
+	return nil
+}
+
+func flattenPhoneNumber(phonenumber *platformclientv2.Phonenumber) []interface{} {
+	phonenumberInterface := make(map[string]interface{})
+	if phonenumber.Display != nil {
+		phonenumberInterface["display"] = *phonenumber.Display
+	}
+	if phonenumber.Extension != nil {
+		phonenumberInterface["extension"] = *phonenumber.Extension
+	}
+	if phonenumber.AcceptsSMS != nil {
+		phonenumberInterface["accepts_sms"] = *phonenumber.AcceptsSMS
+	}
+	if phonenumber.E164 != nil {
+		phonenumberInterface["e164"] = *phonenumber.E164
+	}
+	if phonenumber.CountryCode != nil {
+		phonenumberInterface["country_code"] = *phonenumber.CountryCode
+	}
+	return []interface{}{phonenumberInterface}
+}
+
+func buildSdkAddresse(d *schema.ResourceData, key string) *platformclientv2.Contactaddress {
+	if d.Get(key) != nil {
+		addressData := d.Get(key).([]interface{})
+		if len(addressData) > 0 {
+			addressMap := addressData[0].(map[string]interface{})
+			address1 := addressMap["address1"].(string)
+			address2 := addressMap["address2"].(string)
+			city := addressMap["city"].(string)
+			state := addressMap["state"].(string)
+			postalcode := addressMap["postal_code"].(string)
+			countrycode := addressMap["country_code"].(string)
+
+			return &platformclientv2.Contactaddress{
+				Address1:    &address1,
+				Address2:    &address2,
+				City:        &city,
+				State:       &state,
+				PostalCode:  &postalcode,
+				CountryCode: &countrycode,
+			}
+		}
+	}
+	return nil
+}
+
+func flattenSdkAddress(address platformclientv2.Contactaddress) []interface{} {
+	addressInterface := make(map[string]interface{})
+	if address.Address1 != nil {
+		addressInterface["address1"] = address.Address1
+	}
+	if address.Address2 != nil {
+		addressInterface["address2"] = address.Address2
+	}
+	if address.City != nil {
+		addressInterface["city"] = address.City
+	}
+	if address.State != nil {
+		addressInterface["state"] = address.State
+	}
+	if address.PostalCode != nil {
+		addressInterface["postal_code"] = address.PostalCode
+	}
+	if address.CountryCode != nil {
+		addressInterface["country_code"] = address.CountryCode
+	}
+	return []interface{}{addressInterface}
+}
+
+func buildSdkTwitterId(d *schema.ResourceData, key string) *platformclientv2.Twitterid {
+	if d.Get(key) != nil {
+		twitterData := d.Get(key).([]interface{})
+		if len(twitterData) > 0 {
+			twitterMap := twitterData[0].(map[string]interface{})
+			id := twitterMap["id"].(string)
+			name := twitterMap["name"].(string)
+			screenname := twitterMap["screen_name"].(string)
+			profileurl := twitterMap["profile_url"].(string)
+
+			return &platformclientv2.Twitterid{
+				Id:         &id,
+				Name:       &name,
+				ScreenName: &screenname,
+				ProfileUrl: &profileurl,
+			}
+		}
+	}
+	return nil
+}
+
+func flattenSdkTwitterId(twitterId platformclientv2.Twitterid) []interface{} {
+	twitterInterface := make(map[string]interface{})
+	if twitterId.Id != nil {
+		twitterInterface["id"] = twitterId.Id
```

Comment:
Is this a GUID stored in genesys? Just curious

Reply:
No, it's an optional parameter you can choose when you create a contact, it's not restricted to a GUID format.